teamsyncd/teamsync.cpp

```diff
@@ -27,6 +27,22 @@ TeamSync::TeamSync(DBConnector *db, DBConnector *stateDb, Select *select) :
     m_lagMemberTable(db, APP_LAG_MEMBER_TABLE_NAME),
     m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME)
 {
+    m_nl_sock = nl_socket_alloc();
+    if (!m_nl_sock)
+    {
+        SWSS_LOG_ERROR("Unable to allocate netlink socket");
```

Comment:
I think it's better to crash it. Otherwise your check will not work

Reply:
I had thought before about introducing a throw statement and crashing if the socket_alloc() fails. But didn't do so as I have seen these failure messages of "Unable to initialize team socket" with the team_alloc()/team_init() in other scenario's like back to back config reload command issued multiple times. So didn't want teamsyncd to fail when we try to spawn the process because if this additional check we are introducing. 

But I think on a second thought -- I agree with you, it is good to throw error and crash as this is not a good scenario. Shall update, thanks.